src/snowflake/snowpark/_internal/data_source/drivers/base_driver.py

```diff
@@ -46,6 +47,49 @@ def infer_schema_from_description(self, table_or_query: str) -> StructType:
             cursor.close()
             conn.close()
 
+    def udtf_ingestion(
+        self,
+        session: "snowflake.snowpark.Session",
+        schema: StructType,
+        partition_table: str,
+        external_access_integrations: str,
+        fetch_size: int = 1000,
+    ) -> "snowflake.snowpark.DataFrame":
+        session.udtf.register(
+            self.udtf_class_builder(fetch_size),
+            name="dbapi",
```

Comment:
1. should this be a temp udtf? or we delete the udtf afterwards?
2. we'd better choose unique name for it, which won't conflict users' object
3. the name can be a string constant